src/unitxt/metrics.py

```diff
@@ -804,15 +872,55 @@ def compute(self, references: List[List[Any]], predictions: List[Any]) -> List[A
         return self.pipe(inputs, batch_size=self.batch_size)
 
 
-# Normalized Discounted Cumulative Gain
 class NDCG(GlobalMetric):
+    """
+    Normalized Discounted Cumulative Gain: measures the quality of ranking with respect to ground truth ranking scores.
+
+    As this measures ranking, it is a global metric that can only be calculated over groups of instances. In the
+    common use case where the instances are grouped by different queries, i.e., where the task is to provide a
+    relevance score for a search result w.r.t. a query, an nDCG score is calculated per each query (specified in the
+    "query" input field of an instance) and the final score is the average across all queries.
+    Note that the expected scores are relevance scores (i.e., higher is better) and not rank indices. The absolute
+    value of the scores is only meaningful for the reference scores; for the predictions, only the ordering of the
+    scores affects the outcome - for example, predicted scores of [80, 1, 2] and [0.8, 0.5, 0.6] will receive
+    the same nDCG score w.r.t. a given set of reference scores.
+
+    See also https://en.wikipedia.org/wiki/Discounted_cumulative_gain
+    """
+
     main_score = "nDCG"
 
     def prepare(self):
         from sklearn.metrics import ndcg_score
 
+        super().prepare()
         self.eval = ndcg_score
 
-    def compute(self, references: List[List[float]], predictions: List[float]) -> dict:
-        scores_dict = {self.main_score: self.eval([references], [predictions])}
+    def compute_with_additional_inputs(
+        self, references: List[List[Any]], predictions: List[Any], additional_inputs: List[Any]
```

Comment:
i am not going over this one in depth, i suppose it was written by both you and Ariel, hence i consider it well-reviewed :)

Reply:
Sure...